src/api/types.js

```diff
@@ -1,7 +1,16 @@
 /**
- * @typedef {object} FormConfiguration
- * @property {string} id - The unique ID of the form
+ * @typedef {object} FormConfigurationInput
+ * @property {string} title - The human-readable title of the form
  * @property {string} organisation - The organisation this form belongs to
  * @property {string} teamName - The name of the team who own this form
  * @property {string} teamEmail - The email of the team who own this form
  */
+
+/**
+ * @typedef {object} IdMixin
+ * @property {string} id - The id of the form
+ */
+
+/**
+ * @typedef {FormConfigurationInput & IdMixin} FormConfiguration
```

Comment:
Although it might be easier to add `id` to `FormConfiguration` instead

Then omit it when used as form creation input using `Omit<FormConfiguration, 'id'>`?

Reply:
I'd rather maintain two separate types, bit easier to read IMO. Inline is fine - merged.